.github/workflows/structured-logging-schema-check.yml

```diff
@@ -0,0 +1,69 @@
+# This Action checks makes a dbt run to sample json structured logs
+# and checks that they conform to the currently documented schema.
+# 
+# If this action fails it either means we have unintentionally deviated
+# from our documented structured logging schema, or we need to bump the
+# version of our structured logging and add new documentation to
+# communicate these changes.
+
+
+name: Structured Logging Schema Check
+on:
+  push:
+    branches:
+      - "main"
+      - "*.latest"
+      - "releases/*"
+  pull_request:
+  workflow_dispatch:
+
```

Comment:
I would add `permissions: read-all` like [here](https://github.com/dbt-labs/dbt-core/blob/main/.github/workflows/main.yml#L27). This scope the GitHub token down so you can't write back to the repo